InformationScripting/src/parsing/ArgumentParser.cpp

```diff
@@ -41,21 +42,27 @@ ScopedArgumentQuery::ScopedArgumentQuery(Model::Node* target, std::initializer_l
 	argParser_->setSingleDashWordOptionMode(QCommandLineParser::ParseAsLongOptions);
 
 	if (!argParser_->parse(args))
-		qWarning() << args[0] << "parse failure"; // TODO warn user
+		throw ParsingException("General parsing error");
 
 	QString scope = argParser_->value(SCOPE_ARGUMENT_NAMES[0]);
 	if (scope == "g") scope_ = Scope::Global;
 	else if (scope == "of") scope_ = Scope::Input;
 }
 
-QString ScopedArgumentQuery::argument(const QString& argName) const
+QString ArgumentParser::argument(const QString& argName) const
 {
 	return argParser_->value(argName);
 }
 
-bool ScopedArgumentQuery::isArgumentSet(const QString& argName) const
+bool ArgumentParser::isArgumentSet(const QString& argName) const
 {
 	return argParser_->isSet(argName);
 }
 
+void InformationScripting::ArgumentParser::checkRule(const Arguments::ArgumentRule& rule) const
```

Comment:
This method looks a bit unnecessary. Can't `rule.check` throw itself? Why do we need this wrapper?